charts/k8ssandra/templates/traefik.ingressroutes.yaml

```diff
@@ -43,3 +45,92 @@ spec:
         - name: {{ $clusterName }}-{{ $datacenterName }}-service
           port: 9042
 {{- end }}
+{{- if $traefik.stargate.enabled }}
```

Comment:
Since Stargate supports native K8s ingress it doesn't necessarily have to live under `$traefik.stargate`. Traefik is only used here for TCP ingress routing. I'm open to nesting HTTP-based ingress under a separate key (IE `.Values.ingress.stargate.enabled`).

Separately we should consider what this looks like when Stargate ends up within a sub-chart.

Reply:
Fair point. Maybe we should avoid implementation-specific keys in our config entirely? `ingress.traefik` would then only be for traefik-specific settings.

Reply:
Possibly, we could have a parameter for `ingressProvider` with a value of `traefik`, `nginx`, `kong`, etc. Then in the template match on that value.

FWIW I think we will most likely be moving the ingress configs under each sub-chart for example:

```yaml
kube-prometheus-stack
  grafana:
    ingress:
      enabled: false
      # Uncomment to listen for Grafana traffic under a route prefix
      # path: /grafana
```

so we may end up with something like

```yaml
stargate
  ingress:
    enabled: true
    tcp_ingress_provider: traefik
    host: stargate.example.com
```

What do you think?

Reply:
>FWIW I think we will most likely be moving the ingress configs under each sub-chart for example:

Yes, this is exactly what I have in mind. And we can put common code/templates into k8ssandra-common. 

For now, I wouldn't worry too much about the placement and organization of these properties since it will change when we create a separate Stargate chart.

Reply:
Okay, I'm going to leave this as-is and revise when we make a separate chart.